# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderD3D11TextureHost.h
# Commit: 49ae9de85ab9
# Full Hash: 49ae9de85ab90bf13b9c6bc2c8b6bb540140c90d
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Support using D3D11 textures as external compositor surfaces with D3D11 compositor. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95800
# ==============================================================================

diff -r 91d8f0f829fd -r 49ae9de85ab9 gfx/webrender_bindings/RenderD3D11TextureHost.h
--- a/gfx/webrender_bindings/RenderD3D11TextureHost.h	Thu Nov 05 20:18:13 2020 +0000
+++ b/gfx/webrender_bindings/RenderD3D11TextureHost.h	Thu Nov 05 19:27:23 2020 +0000
@@ -102,9 +102,14 @@
   explicit RenderDXGIYCbCrTextureHost(WindowsHandle (&aHandles)[3],
                                       gfx::YUVColorSpace aYUVColorSpace,
                                       gfx::ColorDepth aColorDepth,
+                                      gfx::ColorRange aColorRange,
                                       gfx::IntSize aSizeY,
                                       gfx::IntSize aSizeCbCr);
 
+  RenderDXGIYCbCrTextureHost* AsRenderDXGIYCbCrTextureHost() override {
+    return this;
+  }
+
   wr::WrExternalImage Lock(uint8_t aChannelIndex, gl::GLContext* aGL,
                            wr::ImageRendering aRendering) override;
   void Unlock() override;
@@ -115,6 +120,8 @@
 
   bool SyncObjectNeeded() override { return true; }
 
+  gfx::ColorRange GetColorRange() const { return mColorRange; }
+
   // RenderTextureHostSWGL
   gfx::SurfaceFormat GetFormat() const override {
     return gfx::SurfaceFormat::YUV;
@@ -131,6 +138,10 @@
   bool EnsureD3D11Texture2D(ID3D11Device* aDevice);
   bool LockInternal();
 
+  ID3D11Texture2D* GetD3D11Texture2D(uint8_t aChannelIndex) {
+    return mTextures[aChannelIndex];
+  }
+
  private:
   virtual ~RenderDXGIYCbCrTextureHost();
 
@@ -156,6 +167,7 @@
 
   gfx::YUVColorSpace mYUVColorSpace;
   gfx::ColorDepth mColorDepth;
+  gfx::ColorRange mColorRange;
   gfx::IntSize mSizeY;
   gfx::IntSize mSizeCbCr;
 